spec/acceptance/tests/yumrepo_spec.rb

```diff
@@ -90,5 +90,65 @@ def resource(host, type, name)
         assert_match(res['descr'], 'Puppet Labs Products El 7 - $basearch')
       end
     end
+
+    describe '`proxy` property' do
+      context 'when set to a URL' do
+        it 'applies idempotently' do
+          pp = <<-MANIFEST
+          yumrepo {'proxied-repo':
+            baseurl => 'http://myownmirror',
+            proxy   => 'http://proxy.example.com:3128',
+          }
+          MANIFEST
+
+          apply_manifest(pp, catch_failures: true)
+          apply_manifest(pp, catch_changes: true)
+        end
+
+        describe file('/etc/yum.repos.d/proxied-repo.repo') do
+          its(:content) { is_expected.to contain('proxy=http://proxy.example.com:3128') }
+        end
+      end
+      context 'when set to `absent`' do
+        it 'applies idempotently' do
+          pp = <<-MANIFEST
+          yumrepo {'proxied-repo':
+            baseurl => 'http://myownmirror',
+            proxy   => absent,
+          }
+          MANIFEST
+
+          apply_manifest(pp, catch_failures: true)
+          apply_manifest(pp, catch_changes: true)
+        end
+
+        describe file('/etc/yum.repos.d/proxied-repo.repo') do
+          its(:content) { is_expected.not_to contain('proxy') }
+        end
+      end
+      context 'when set to `_none_`' do
+        it 'applies idempotently' do
+          pp = <<-MANIFEST
+          yumrepo {'proxied-repo':
+            baseurl => 'http://myownmirror',
+            proxy   => '_none_',
+          }
+          MANIFEST
+
+          apply_manifest(pp, catch_failures: true)
+          apply_manifest(pp, catch_changes: true)
+        end
+
+        if fact('os.release.major') == '8'
```

Comment:
As much as I can be, I think this test should execute correctly.  beaker-rspec doesn't appear to support EL8 though.

Reply:
> NameError:
  uninitialized constant Beaker::HostPrebuiltSteps::RHEL8_PACKAGES
  Did you mean?  Beaker::HostPrebuiltSteps::SLES_PACKAGES

Reply:
Seems that this module is still on beaker 3.
`RHEL8_PACKAGES` were added in beaker 4

Reply:
Good spot.  I've tweaking the Gemfile and spec helper locally, and the tests do run on EL8.  Probably not much point me committing any of these changes though as you'll prefer to migrate to litmus? (and if the tests could run in travis, that'd be helpful too).

Reply:
This test is currently failing on Fedora.
Can you please change the condition from `==8` to `>=8`.

More info about this in the conversation of this PR: https://github.com/puppetlabs/puppetlabs-yumrepo_core/pull/22/files#diff-dc408e2d28dac43c3fdd4c2012e8d74bR314